tools/hubl/internal/keyring/keyring.go

```diff
@@ -0,0 +1,113 @@
+package keyring
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"path"
+
+	"github.com/spf13/cobra"
+
+	_ "cosmossdk.io/api/cosmos/crypto/ed25519"
+	_ "cosmossdk.io/api/cosmos/crypto/secp256k1"
+	_ "cosmossdk.io/api/cosmos/crypto/secp256r1"
+	"cosmossdk.io/tools/hubl/internal/config"
+	"cosmossdk.io/tools/hubl/internal/flags"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/keys"
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	cryptocodec "github.com/cosmos/cosmos-sdk/crypto/codec"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func Cmd(chainName string) *cobra.Command {
+	shortDesc := fmt.Sprintf("Keyring management for %s", chainName)
+	if chainName == "" {
+		chainName = config.GlobalKeyringDirName
+		shortDesc = "Global keyring management for Hubl"
+	}
+
+	keyringCmd := &cobra.Command{
+		Use:   "keys",
+		Short: shortDesc,
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			registry := codectypes.NewInterfaceRegistry()
+			cryptocodec.RegisterInterfaces(registry)
+			cdc := codec.NewProtoCodec(registry)
+
+			configDir, err := config.GetConfigDir()
+			if err != nil {
+				return err
+			}
+
+			cfg, err := config.Load(configDir)
+			if err != nil {
+				return err
+			}
+
+			backend, err := cfg.GetKeyringBackend(chainName)
+			if err != nil {
+				return err
+			}
+
+			if changed := cmd.Flags().Changed(flags.FlagKeyringBackend); changed {
+				b, err := cmd.Flags().GetString(flags.FlagKeyringBackend)
+				if err != nil {
+					return err
+				}
+
+				backend = b
+			}
+
+			// set global bech32 prefix
+			// TODO remove bech32 prefix from client/keys
```

Comment:
Doing this now.

Reply:
Global bleeds everywhere in client/keys. Best to way for keyring v2.

Reply:
https://github.com/cosmos/cosmos-sdk/pull/17503 will allow to just use the builder address codec.

Reply:
The follow-up got merged first lol -> https://github.com/cosmos/cosmos-sdk/pull/17503
Done then!